File: PlayerUI/ReactNative/swift/PlayerReactNative/BCOVVideoPlayerManager.m
//
//  BCOVVideoPlayerManager.m
//  PlayerReactNative
//
//  Created by Carlos Ceja.
//

#import <React/RCTBridge.h>
#import <React/RCTUIManager.h>

#import "PlayerReactNative-Swift.h"

#import "BCOVVideoPlayerManager.h"


@implementation BCOVVideoPlayerManager

RCT_EXPORT_MODULE(BCOVVideoPlayer)

- (UIView *)view
{
    BCOVVideoPlayer *player = [[BCOVVideoPlayer alloc] initWithFrame:UIScreen.mainScreen.bounds];
    player.eventDispatcher = (RCTEventDispatcher *)self.bridge.eventDispatcher;
    return player;
}

- (dispatch_queue_t)methodQueue
{
    return self.bridge.uiManager.methodQueue;
}

RCT_EXPORT_VIEW_PROPERTY(options, NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(onReady, RCTDirectEventBlock);
RCT_EXPORT_VIEW_PROPERTY(onProgress, RCTDirectEventBlock);
RCT_EXPORT_METHOD(playPause:(nonnull NSNumber *)reactTag isPlaying:(BOOL)isPlaying {
    [self.bridge.uiManager addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
        BCOVVideoPlayer *player = (BCOVVideoPlayer *)viewRegistry[reactTag];
        if ([player isKindOfClass:[BCOVVideoPlayer class]])
        {
            [player playPause:isPlaying];
        }
    }];
});

@end

File: PlayerUI/ReactNative/swift/PlayerReactNative/BCOVVideoPlayerManager.swift
//
//  BCOVVideoPlayerManager.swift
//  PlayerReactNative
//
//  Created by Carlos Ceja.
//

import Foundation
import UIKit

import React

@objc(BCOVVideoPlayerManager)
class BCOVVideoPlayerManager: RCTViewManager {

    @objc 
    override func view() -> UIView! {
        return BCOVVideoPlayer(eventDispatcher: self.bridge.eventDispatcher())
    }
    
    func methodQueue() -> DispatchQueue {
        return self.bridge.uiManager.methodQueue
    }
    
    override static func requiresMainQueueSetup() -> Bool {
      return true
    }
    
    @objc func updateFromManager(_ node: NSNumber, count: NSNumber) {
      DispatchQueue.main.async {
//        let component = self.bridge.uiManager.view(
//          forReactTag: node
//        ) as! CounterView
//        component.update(value: count)
      }
    }

}
